File: Android/app/src/main/java/com/edu/primary/database/entity/AnswerEntity.java
package com.edu.primary.database.entity;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "answers")
public class AnswerEntity {
    @PrimaryKey(autoGenerate = true)
    public long id;

    public long userId;
    public long questionId;
    public String userAnswer;
    public boolean isCorrect;
    public long answerTime;

    public AnswerEntity() {
    }

    @Ignore
    public AnswerEntity(long userId, long questionId, String userAnswer, boolean isCorrect) {
        this.userId = userId;
        this.questionId = questionId;
        this.userAnswer = userAnswer;
        this.isCorrect = isCorrect;
        this.answerTime = System.currentTimeMillis();
    }
}

File: Android-1/app/src/main/java/com/edu/primary/data/local/entity/AnswerEntity.kt
package com.edu.primary.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "answers")
data class AnswerEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val userId: Long,
    val questionId: Long,
    val userAnswer: String,
    val isCorrect: Boolean,
    val answerTime: Long = System.currentTimeMillis()
)
